mewarpx/mewarpx/util.py

```diff
@@ -30,3 +31,18 @@ def init_libwarpx(ndim, rz):
     # This just quiets linters like pyflakes by using the otherwise-unused
     # variable
     assert pywarpx._libwarpx
+
+
+# https://stackoverflow.com/questions/600268/mkdir-p-functionality-in-python
+def mkdir_p(path):
+    """Make directory and parent directories if they don't exist.
+
+    Do not throw error if all directories already exist.
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
```

Comment:
It looks like if the test directory with the same name already exists, you'll just keep using that directory. Would it be safer to truncate that directory in case the previously ran test had a file that won't be over written? Another alternative is to call a method that would re-generate and broadcast a new test directory name.

Reply:
I actually don't worry about collisions - the function is written this way primarily so that if `test_files` directory didn't exist it will be created rather than erroring out. Though I think `temp_files`  is a better name, so I just changed that... We've used it in other cases like making folders for diagnostics to go in; you don't have to worry if the parent directory of a diagnostic does or doesn't exist.